diff --git a/haskell/cc.bzl b/haskell/cc.bzl
index 784d25a0..789e8628 100644
--- a/haskell/cc.bzl
+++ b/haskell/cc.bzl
@@ -199,6 +199,8 @@ def ghc_cc_program_args(hs, cc):
         cc,
         "-pgml",
         cc,
+        "-pgmlc",
+        cc,
         # Setting -pgm* flags explicitly has the unfortunate side effect
         # of resetting any program flags in the GHC settings file. So we
         # restore them here. See
diff --git a/haskell/private/cabal_wrapper.py b/haskell/private/cabal_wrapper.py
index 222f381b..7124630f 100755
--- a/haskell/private/cabal_wrapper.py
+++ b/haskell/private/cabal_wrapper.py
@@ -218,7 +218,8 @@ with tmpdir() as distdir:
     runghc_args = [arg.replace("./", execroot + "/") for arg in runghc_args]
     run([runghc] + runghc_args + [setup, "configure", \
         component, \
-        "--verbose=0", \
+        "--verbose=3", \
+        "--ghc-option=-v3", \
         "--user", \
         "--with-compiler=" + ghc,
         "--with-hc-pkg=" + ghc_pkg,
